gnovm/stdlibs/std/banker.go

```diff
@@ -44,12 +44,18 @@ func X_bankerSendCoins(m *gno.Machine, bt uint8, fromS, toS string, denoms []str
 	amt := CompactCoins(denoms, amounts)
 	from, to := crypto.Bech32Address(fromS), crypto.Bech32Address(toS)
 
+	pkgAddr := ctx.OrigPkgAddr
+	if m.Realm != nil {
+		pkgPath := m.Realm.Path
+		pkgAddr = gno.DerivePkgAddr(pkgPath).Bech32()
+	}
+
 	if bt == btOrigSend || bt == btRealmSend {
-		if from != ctx.OrigPkgAddr {
+		if from != pkgAddr {
```

Comment:
The earlier code checks against the package address for SendCoins() in the OrigSendBanker and RealmSendBanker separately.

https://github.com/gnolang/gno/blob/6c5b4cfd6ab12838ba86c4913613b8cc35cfe046/gnovm/stdlibs/std/banker.go#L97
https://github.com/gnolang/gno/blob/6c5b4cfd6ab12838ba86c4913613b8cc35cfe046/gnovm/stdlibs/std/banker.go#L144

The code is the same, but the state of the package address in different banker instances could be different. In the current code, it is merged into one logic, where it checks either OrigPkgAddr or m.Realm.Path for SendCoin() of both  OrigSendBanker and RealmSendBanker together. However, these should be checked separately.  If not, other places that require checking OriginSendBanker's OrigPkgAddr might fail when ctx.OrignPkgAdd and m.Realm.PathAddress have different values.

Reply:
looking into this
(seems this logic is also related to your another comments about test cases).

Reply:
UPDATE //
I kinda missed internal behavior of OrigSend banker. 
I'll divide logic and testcases
Thanks for pointing this out